Makefile

```diff
@@ -1,33 +1,61 @@
 COQMODULE    := OpencypherCoq
 COQTHEORIES  := src/*.v
 
-.PHONY: all theories clean tounicode
+.PHONY: all theories fmt lint docs clean tounicode
 
 all: build
 
 build: Makefile.coq
+	eval $$(opam env) && \
 	$(MAKE) -f Makefile.coq all
 
 quick: Makefile.coq
+	eval $$(opam env) && \
 	$(MAKE) -f Makefile.coq quick
 
 quick-check: Makefile.coq
+	eval $$(opam env) && \
 	$(MAKE) -f Makefile.coq vio2vo J=6
 
-Makefile.coq: Makefile $(COQTHEORIES)
+_CoqProject.dune: Makefile $(COQTHEORIES)
+	(echo "-R _build/default/src $(COQMODULE)"; \
+	echo _build/default/$(COQTHEORIES)) > _CoqProject.dune
+
+_CoqProject: Makefile $(COQTHEORIES)
 	(echo "-R src $(COQMODULE)"; \
 	echo $(COQTHEORIES)) > _CoqProject
-	coq_makefile -f _CoqProject -o Makefile.coq
+
+Makefile.coq: _CoqProject
+	opam exec -- coq_makefile -f _CoqProject -o Makefile.coq
 
 %.vo: Makefile.coq
+	eval $$(opam env) && \
 	$(MAKE) -f Makefile.coq "$@"
 
 %.vio: Makefile.coq
+	eval $$(opam env) && \
 	$(MAKE) -f Makefile.coq "$@"
 
 clean:
+	rm -f src/*.glob
+	rm -f src/*.vo
+	rm -f src/*.vok
+	rm -f src/*.vos
+	rm -f src/.*.aux
+	rm -f _CoqProject.dune
```

Comment:
Why do you need this? Isn't `$(MAKE) -f Makefile.coq clean` doing that?

Reply:
It doesn't remove `.aux` files. But you are right that it removes all the others. Fixed this.